### PR TITLE
Commitment task + HotShot contract deploy script

### PR DESCRIPTION
**Pull request by nomaxg**
Added a simple script to launch the HotShot contract and run the commitment task. This is useful for integration demos.

### CODE REVIEW

File: docker/commitment-task.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu:jammy
+
+ARG TARGETARCH
+
+RUN apt-get update &&  rm -rf /var/lib/apt/lists/*
```
Comment (sveitser) [inline docker/commitment-task.Dockerfile:5]: I think there's no need for this unless we're installing packages with apt in the Dockerfile.

File: sequencer/src/bin/commitment-task.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        provider.set_interval(Duration::from_millis(10));
+        let test_system = TestL1System::deploy(provider).await.unwrap();
+        tracing::info!(
+            "HotShot Contract launched at address: {:?}",
+            test_system.hotshot.address()
+        );
+    }
+
+    tracing::info!("Launching HotShot commitment task..");
+    run_hotshot_commitment_task(&hotshot_contract_options).await;
```
Comment (sveitser) [inline sequencer/src/bin/commitment-task.rs:67]: It looks like we are not setting `hotshot_address` after deploying the contract.

File: sequencer/src/bin/commitment-task.rs [hunk 3]
```diff
@@ -13,28 +12,29 @@ pub struct Options {
...
 
     /// URL of layer 1 Ethereum JSON-RPC provider.
-    #[clap(long, env = "ESPRESSO_DEMO_L1_PROVIDER")]
+    #[clap(long, env = "ESPRESSO_SEQUENCER_L1_PROVIDER")]
     pub l1_provider: Url,
 
-    /// Address of HotShot contract on layer 1.
-    #[clap(long, env = "ESPRESSO_DEMO_HOTSHOT_ADDRESS")]
-    pub hotshot_address: Address,
+    /// Optional address of a predeployed HotShot contract on layer 1. If unspecified, the task will deploy
```
[the remark this answers is not in the document]
Comment (jbearer) [reply sequencer/src/bin/commitment-task.rs:18]: Yeah I think actually having the `--deploy` flag is a good idea even though it's redundant and we only check it if `hotshot_address` is `None`

[the remark this answers is not in the document]
Comment (nomaxg) [reply sequencer/src/bin/commitment-task.rs:18]: make sense, I'll add deploy back

**Review by jbearer — APPROVED** [review]

**Pull request merged**